Decorator/GuzzleClientFactoryDecorator.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Auxmoney\OpentracingGuzzleBundle\Decorator;
+
+use Auxmoney\OpentracingGuzzleBundle\Middleware\GuzzleRequestSpanning;
+use Auxmoney\OpentracingGuzzleBundle\Middleware\GuzzleTracingHeaderInjection;
+use GuzzleHttp\Client;
+use GuzzleHttp\HandlerStack;
+use Symfony\Component\DependencyInjection\Reference;
+
+final class GuzzleClientFactoryDecorator
+{
+
+    public static function decorate(
+        Client $client,
+        GuzzleRequestSpanning $spanning,
+        GuzzleTracingHeaderInjection $headerInjection
+    )
+    {
+        $handler = $client->getConfig("handler");
+        if ($handler instanceof HandlerStack) {
```

Comment:
Why is checking `instanceof HandlerStack` needed ?

Reply:
i dont find strong typing, so i'm just carefull, i think it's better to not add the span than broke the code

Reply:
I don't think it ca be anything else than a `HandlerStack` but ok, it shouldn't hurt.
Maybe add a warning log if it's not.

Reply:
It could be a `callable`, according to the [documentation](https://github.com/guzzle/guzzle/blob/6.5/src/Client.php#L46-L54)